backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/settings/ModularContractSettingsBL.java

```diff
@@ -133,6 +136,36 @@ private void createInformativeLogsModule(@NonNull final ModularContractSettingsI
 		);
 	}
 
+	private void createDefinitiveInvoiceModule(@NonNull final ModularContractSettingsId modularContractSettingsId)
+	{
+		final ModularContractSettings modularContractSettings = getById(modularContractSettingsId);
+		final ProductId moduleProductId;
+		final ModularContractType moduleContractType;
+		final ProductId processedProductId = modularContractSettings.getProcessedProductId();
+		if(processedProductId != null)
+		{
+			moduleProductId = processedProductId;
+			moduleContractType = modularContractSettingsDAO.getContractTypeById(ModularContract_Constants.CONTRACT_MODULE_TYPE_DefinitiveInvoiceProcessedProduct);
+		}
+		else
+		{
+			moduleProductId = modularContractSettings.getRawProductId();
+			moduleContractType = modularContractSettingsDAO.getContractTypeById(ModularContract_Constants.CONTRACT_MODULE_TYPE_DefinitiveInvoiceRawProduct);
+		}
+
+		final String moduleName = TranslatableStrings.adElementOrMessage("DefinitiveInvoice").translate(Env.getADLanguageOrBaseLanguage());
```

Comment:
pls extract DefinitiveInvoice as AdMessageKey constant

Reply:
An AD_Element already exists for this, see https://github.com/metasfresh/metasfresh/blob/6127488cb290c65b730d91e2233e5db997619ebb/backend/de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5723960_sys_DefinitiveInvoice_Element.sql#L2
The most I can do for it is extract it as constant in the class.